src/session_pool/session_pool.js

```diff
@@ -228,15 +261,17 @@ export class SessionPool extends EventEmitter {
     }
 
     /**
-     * Removes `Session` instance from `SessionPool`.
-     * @param {Session} session  - Session to be removed
+     * Removes retired `Session` instances from `SessionPool`.
      * @private
      */
-    _removeSession(session) {
-        const sessionIndex = this.sessions.findIndex((storedSession) => storedSession.id === session.id);
+    _removeRetiredSessions() {
```

Comment:
@petrpatek sorry, I don't get the first one. `session_${cryptoRandomObjectId(10)}` is the default value assigned to the newly created session, I am not sure why I can't pass the `id` explicitly? And it's working with no issues...

Regarding the second comment - but I do remove the session only by reaching retired stated. The previous session pool would remove only 1 retired session and replace it with a new one, this way - the session pool clears all retired sessions at once, creates a new one, and leaves more space for the new sessions. Basically - I don't change anything related to the events, I only change whether it's 1 already retired session is removed or all of them.

Reply:
I've just re-read the message and realized that I was thinking about a completely different thing yesterday...

On one hand it makes sense - since the session is not usable at this point and to erase it sessionPool is basically waiting to pick the retired session and then remove it from the pool.

On the other hand - it kinda seems like session pool should manipulate sessions (add/remove them), but not session should remove itself from the pool?

Reply:
I totally agree that the session pool should manage to remove/adding sessions. This way, the session pool would listen to the emitted event by session and then removed it, but it was just an idea. This way, it works too 👌